crates/superposition_types/src/lib.rs

```diff
@@ -243,9 +244,92 @@ impl Display for RegexEnum {
     }
 }
 
-#[derive(Clone, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum HeadersEnum {
+    ConfigVersion,
+}
+
+impl fmt::Display for HeadersEnum {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            HeadersEnum::ConfigVersion => write!(f, "x-config-version"),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum HttpMethod {
+    Get,
+    Post,
+    Put,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Authorization {
+    pub key: String,
+    pub value: String,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Webhook {
+    pub url: String,
+    pub method: HttpMethod,
+    pub headers: Vec<HeadersEnum>,
+    pub authorization: Authorization,
+}
+
+#[derive(Serialize, Deserialize)]
+pub enum WebhookEvent {
```

Comment:
Why is this not usable for CAC, the webhook object will be similar, it can be reused. If it can't be reused then it can be renamed then. But now i think this is fine